### PR TITLE
Make SerialVersionCalc.java pass all checks on the spotbugs branch

### PR DESCRIPTION
**Pull request by proski**
Reduce complexity of uuid() by moving code to private methods. Make
comparators serializable. Remove unused import.

### CODE REVIEW

**Comment by proski** [comment]
I was looking for the jar file of the latest plugin for spotbugs to use with Eclipse. I could only find the one for findbugs. So I decided to build one for myself. I downloaded the source and used `mvn clean install` as described in `README.md`. I know it's under "Setting up for Development", but I didn't see instructions how to compile the code.

I got multiple errors, so I just gave up for a while until I found some time today. I was able to fix the issues. Then it turned out I need to submit patched against the findbugs branch. There are some differences between the branches, and I don't understand if they are needed. I was able to compile the code from the spotbugs branch on the findbugs branch with no changes.

Anyway, I backported my change and preserved the differences between the branches. One more fix will be needed on the spotbugs branch. The `FieldOrMethod` import is unnecessary and should be removed.

**Comment by proski** [comment]
Would it be possible to treat spotbugs as the primary development branch? Asking every potential contributor to backport their code to the findbugs branch means that backporting is done by people who don't know what they should be avoiding (e.g. Java 8 specific constructs) or don't have tools to test their backport (e.g. older Java versions). Backporting is better done by someone who knows exactly what to look for.

I would also appreciate if the continuous integration could be fixed and extended to test all relevant branched to the highest standards (e.g. checking both branches with findbugs and spotbugs). It's a huge turnoff for potential contributors when the automated builds fail.

I'm sure the project would see more contributions if it makes it easier and safer to add new code.

File: src/main/java/com/mebigfatguy/fbcontrib/utils/SerialVersionCalc.java [hunk 1]
```diff
@@ -40,71 +41,80 @@
...
+        }
+    }
+
+    private static void writeFields(DataOutput out, JavaClass cls) throws IOException {
+        Field[] fields = cls.getFields();
+        if (fields.length > 0) {
+            fields = fields.clone();
+            Arrays.sort(fields, new FieldSorter());
+            for (Field field : fields) {
+                if (!field.isPrivate() || (!field.isStatic() && !field.isTransient())) {
```
Comment (ThrawnCA) [inline src/main/java/com/mebigfatguy/fbcontrib/utils/SerialVersionCalc.java:61]: So...this outputs all public/protected/package fields, plus private non-transient instance fields? I have to think there's a simpler way to express that.

**Review by ThrawnCA — CHANGES_REQUESTED** [review]

**Comment by mebigfatguy** [comment]
thanks!

**Pull request merged**